validator/validate.js

```diff
@@ -0,0 +1,470 @@
+    type: 'array'
+  },
+  "w" : {
+    alias: 'fiware:warnings',
+    describe: 'How to handle FIWARE Data Models checks warnings.\n true (default) - print warnings, but does not fail. \n ignore -  do nothing and do not print warnings.\n fail - print warnings, and fails.',
+    type: 'string'
+  },
+  "p" : {
+    alias: 'fiware:path',
+    describe: 'The path of FIWARE Data Model(s) to be validated (if recursion enabled, it will be the starting point of recursion)',
+    demand: false,
+    type: 'string'
+  },
+  "h" : {
+    alias: 'help',
+    describe: 'Print the help message',
+    demand: false
+  }
+},"Usage: validate -p DataModel -w ignore -i [common-schema.json,geometry-schema.json]").file('config.json');
+
+if (nconf.get('fiware:importSchemas') == null) {
+  nconf.set('fiware:importSchemas', ['common-schema.json','geometry-schema.json']);
+}
+if (nconf.get('fiware:warnings') == null) {
+  nconf.set('fiware:warnings', 'true');
+}
+if (nconf.get('fiware:warningChecks') == null) {
+  nconf.set('fiware:warningChecks', ['schemaExist','docExist','docFolderExist','exampleExist','modelNameValid','readmeExist']);
+}
+if (nconf.get('fiware:recursiveScan') == null) {
+  nconf.set('fiware:recursiveScan', true);
+}
+if (nconf.get('fiware:validateExamples') == null) {
+  nconf.set('fiware:validateExamples', true);
+}
+if (nconf.get('fiware:loadModelCommonSchemas') == null) {
+  nconf.set('fiware:loadModelCommonSchemas', true);
+}
+if (nconf.get('fiware:importExternalSchemaFolders') == null) {
+  nconf.set('fiware:importExternalSchemaFolders', true);
+}
+if (nconf.get('fiware:resolveRemoteSchemas') == null) {
+  nconf.set('fiware:resolveRemoteSchemas', false);
+}
+if (nconf.get('fiware:ignoreFolders') == null) {
+  nconf.set('fiware:ignoreFolders', ['harvest','auxiliary']);
+}
+if (nconf.get('fiware:docFolders') == null) {
+  nconf.set('fiware:docFolders', ['doc']);
+}
+if (nconf.get('fiware:externalSchemaFolders') == null) {
+  nconf.set('fiware:externalSchemaFolders', ['externalSchema']);
+}
+if (nconf.get('ajv:missingRefs') == null) {
+  nconf.set('ajv:missingRefs', 'true');
+}
+if (nconf.get('ajv:extendRefs') == null) {
+  nconf.set('ajv:extendRefs', 'fail');
+}
+if (nconf.get('ajv:allErrors') == null) {
+  nconf.set('ajv:allErrors', true);
+}
+
+/* print help */
+if (nconf.get('h')){
+  nconf.stores.argv.showHelp();
+  return;
+}
+
+/* Check configuration */
+try {
+  nconf.required(['fiware:path']);
+} catch (err) {
+  process.exitCode = -1;
+  console.error("\n Invalid Configuration:"+err.message+"\n");
+  nconf.stores.argv.showHelp();
+  return;
+}
+
+/* Check if path is valid */
+try {
+  // Query the entry
+  stats = fs.lstatSync(nconf.get('fiware:path'));
+
+  // Is it a directory?
+  if (!stats.isDirectory()) {
+    throw new Error("The path passed must be a directory");
+  }
+}
+catch (err) {
+  process.exitCode = -1;
+  console.error("\n Invalid Path: "+err.message+"\n");
+  return;
+}
+
```

Comment:
too long module. this would need to be properly restructured into Node modules and have a clear entry point